python/lsst/sims/alertsim/catalogs/diaObjectCommons.py

```diff
@@ -1,179 +1,401 @@
+
+    _seed = None
+    _rng =None
+
+    column_outputs = ['diaObjectId', 'ra', 'decl', 'ra_decl_Cov', 'radecTai', 'pmRa', 'pmDecl', 'parallax', 'pm_parallax_Cov', 'pmParallaxLnL', 'pmParallaxChi2', 'pmParallaxNdata', 'uPSFluxMean', 'uPSFluxMeanErr', 'uPSFluxSigma', 'uPSFluxChi2', 'uPSFluxNdata', 'gPSFluxMean', 'gPSFluxMeanErr', 'gPSFluxSigma', 'gPSFluxChi2', 'gPSFluxNdata', 'rPSFluxMean', 'rPSFluxMeanErr', 'rPSFluxSigma', 'rPSFluxChi2', 'rPSFluxNdata', 'iPSFluxMean', 'iPSFluxMeanErr', 'iPSFluxSigma', 'iPSFluxChi2', 'iPSFluxNdata', 'zPSFluxMean', 'zPSFluxMeanErr', 'zPSFluxSigma', 'zPSFluxChi2', 'zPSFluxNdata', 'yPSFluxMean', 'yPSFluxMeanErr', 'yPSFluxSigma', 'yPSFluxChi2', 'yPSFluxNdata', 'uFPFluxMean', 'uFPFluxMeanErr', 'uFPFluxSigma', 'gFPFluxMean', 'gFPFluxMeanErr', 'gFPFluxSigma', 'rFPFluxMean', 'rFPFluxMeanErr', 'rFPFluxSigma', 'iFPFluxMean', 'iFPFluxMeanErr', 'iFPFluxSigma', 'zFPFluxMean', 'zFPFluxMeanErr', 'zFPFluxSigma', 'yFPFluxMean', 'yFPFluxMeanErr', 'yFPFluxSigma', 'uLcPeriodic', 'gLcPeriodic', 'rLcPeriodic', 'iLcPeriodic', 'zLcPeriodic', 'yLcPeriodic', 'uLcNonPeriodic', 'gLcNonPeriodic', 'rLcNonPeriodic', 'iLcNonPeriodic', 'zLcNonPeriodic', 'yLcNonPeriodic', 'nearbyObj1', 'nearbyObj1Dist', 'nearbyObj1LnP', 'nearbyObj2', 'nearbyObj2Dist', 'nearbyObj2LnP', 'nearbyObj3', 'nearbyObj3Dist', 'nearbyObj3LnP', 'flags']
+
+    # UCD's - TODO
+
+    ucds = ['', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '']
+
+    # Datatypes as stated in DPDD TODO
+
+    datatypes = ['', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '']
+
+    # Units as stated in DPDD TODO
+
+    units = ['', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '']
+
+    # DIASource attributes with randomly assigned values (for the time being)
+
+    def write_catalog(self, *args, **kwargs):
+        raise NotImplementedError("You cannot call write_catalog() on "
+                                  "VariableStarsDia; write_catalog() does not "
+                                  "know how to deal with the nested structure "
+                                  "of the DIASource schema")
+
+    camera = LsstSimMapper().camera  # the software representation of the LSST camera
```

Comment:
There is now a method `lsst_camera` in `sims_coordUtils` that returns an LSST camera and keeps it in a global cache to prevent it from being re-instantiated needlessly.  You can use it like:
```
from lsst.sims.coordUtils import lsst_camera
camera = lsst_camera()
```

Reply:
done, in diaSourceCommons too